### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes

This PR adds the following features/fixes:
- New `orchestrator_pod_settings` for the `KubernetesOrchestrator`: These allow users to specify pod settings for the orchestrator pod which is responsible for launching the other pods that execute the pipeline steps.
- Allow specifying pod settings via the CLI: In some other stack components, a custom pydantic validator was used to convert json strings coming from the CLI to the correct type needed in the configuration. This was missing for the pod settings on all of our orchestrators that support them. This PR adds a generic method on the `StackComponentConfig` class that solves this issue. We will need to port this to pydantic v2 (FYI @bcdurak)
- Fix the links for pipeline runs and models when connected to a cloud tenant: Changes in the last release led to an issue that currently we log links to the legacy dashboard instead of the cloud dashboard for pipeline runs and models.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by AlexejPenner — APPROVED** [review]
The only thing missing are docs on the KubernetesOrchestrator to describe the difference between the two pod settings - feel free to do that separately from this PR in our docs overhaul branch

**Comment by schustmi** [comment]
@AlexejPenner I added it to the docs on your branch

**Comment by AlexejPenner** [comment]
LGTM then

File: src/zenml/stack/stack_component.py [hunk 1]
```diff
@@ -242,6 +247,35 @@ def _is_part_of_active_stack(self) -> bool:
...
+            value: The value to convert.
+
+        Returns:
+            The converted value.
+
+        Raises:
+            ValueError: If the value is an invalid JSON string.
+        """
+        if isinstance(value, str):
+            if get_origin(field.outer_type_) in [
```
Comment (bcdurak) [inline src/zenml/stack/stack_component.py:264]: Just a small reminder, aside from the `get_origin`, there are a few more things that need to change here when we switch to pydantic v2 such as the `type_`, `outer_type_`, `parse_raw` and `dict()`

File: src/zenml/utils/dashboard_utils.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def is_cloud_server(server_info: ServerModel) -> bool:
     """
     return (
         "organization_id" in server_info.metadata
-        and "cloud.zenml.io" in server_info.server_url
+        and "cloud.zenml.io" in server_info.dashboard_url
```
Comment (bcdurak) [inline src/zenml/utils/dashboard_utils.py:45]: Just out of curiosity, why did we have to change this?

**Review by bcdurak — APPROVED** [review]
LGTM!

File: src/zenml/utils/dashboard_utils.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/utils/dashboard_utils.py:45]: This PR (https://github.com/zenml-io/zenml/pull/2722) essentially switched the URLs of these fields, which means in the latest release `is_cloud_server` is always `False` even for cloud tenants.

**Pull request merged**